document/core/exec/instructions.rst

```diff
@@ -182,6 +182,489 @@ Where the underlying operators are non-deterministic, because they may return on
+
+
+.. _exec-simd-replace-lane:
+
+:math:`\shape\K{.}\REPLACELANE~\laneidx`
+........................................
+
+1. Assert: due to :ref:`validation <valid-simd-replace-lane>`, :math:`\laneidx < \lanes(\shape)`.
+
+2. Let :math:`t_1` be the type :math:`\unpacked(\shape)`.
+
+3. Assert: due to :ref:`validation <valid-simd-replace-lane>`, a value of :ref:`value type <syntax-valtype>` :math:`t_1` is on the top of the stack.
+
+4. Pop the value :math:`t_1.\CONST~v_1` from the stack.
+
+5. Assert: due to :ref:`validation <valid-simd-replace-lane>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+6. Pop the value :math:`\V128.\CONST~c_2` from the stack.
+
+7. Let :math:`n` be the integer :math:`\lanes(\shape)`.
+
+8. Let :math:`i_2^n` be the sequence :math:`\simdto_{\shape}(c_2)`.
+
+9. Let :math:`c` be the result of computing :math:`\simdof_{\shape}(i_2^n \with [\laneidx] = v_1)`
+
+10. Push :math:`\V128.\CONST~c` on the stack.
+
+.. math::
+   \begin{array}{l}
+   \begin{array}{lcl@{\qquad}l}
+   (t_1\K{.}\CONST~v_1)~(\V128\K{.}\CONST~c_2)~\V128\K{.}\REPLACELANE~\laneidx &\stepto& (\V128\K{.}\CONST~c)
+   \end{array}
+   \\ \qquad
+     \begin{array}[t]{@{}r@{~}l@{}}
+      (\iff & n = \lanes(\shape) \\
+       \wedge & i_2^n = \simdto_{\shape}(c_2)) \\
+       \wedge & c = \simdof_{\shape}(i_2^n \with [\laneidx] = v_1)
+     \end{array}
+   \end{array}
+
+
+.. _exec-simd-vunop:
+
+:math:`\shape\K{.}\vunop`
+.........................
+
+1. Assert: due to :ref:`validation <valid-vunop>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+3. Let :math:`c` be the result of computing :math:`\vunop_{\shape}(c_1)`.
+
+4. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~\V128\K{.}\vunop &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff c = \vunop_{\shape}(c_1))
+   \end{array}
+
+
+.. _exec-simd-vbinnop:
+
+:math:`\shape\K{.}\vbinop`
+..........................
+
+1. Assert: due to :ref:`validation <valid-vbinop>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+2. Pop the value :math:`\V128.\CONST~c_2` from the stack.
+
+4. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+4. If :math:`\vbinop_{\shape}(c_1, c_2)` is defined:
+
+   a. Let :math:`c` be a possible result of computing :math:`\vbinop_{\shape}(c_1, c_2)`.
+
+   b. Push the value :math:`\V128.\CONST~c` to the stack.
+
+5. Else:
+
+   a. Trap.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~(\V128\K{.}\CONST~c_2)~\V128\K{.}\vbinop &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff c \in \vbinop_{\shape}(c_1, c_2)) \\
+   (\V128\K{.}\CONST~c_1)~(\V128\K{.}\CONST~c_2)~\V128\K{.}\vbinop &\stepto& \TRAP
+     & (\iff \vbinop_{\shape}(c_1, c_2) = \{\})
+   \end{array}
+
+
+.. _exec-simd-vshiftop:
+
+:math:`t\K{x}N\K{.}\vshiftop`
```

Comment:
why not `shape` here?

Reply:
I wanted the `t` and the `N` to specify the subscript to `vshiftop`. `shape` will work here too, then I will need `\lanes(\shape)` and `\packed(\shape)` instead of `t` and `N`.

Reply:
Yeah, looks fine to me either way, I just thought you were using `shape` below

Reply:
Yea that was a mistake, i fixed them to use `txN`.